### PR TITLE
MODOAIPMH-71: Handle json parsing exceptions gracefully

### PR DESCRIPTION
**Pull request by IlliaDaliek**
Jira: https://issues.folio.org/browse/MODOAIPMH-71

### PURPOSE
Add error handling for both invalid JSON srs response and JSON srs response which cannot be converted to XML representation.

### APPROACH
Catch DecodeException when parsing response to vertx JsonObject, catch the Exception which is thrown when json with incorrect marc structure is converted to XML .
As well small refactoring has been done with removing code duplications.

### CODE REVIEW

**Comment by IlliaDaliek** [comment]
Will fix the sonar code smells later. There the huge ''buildRecords(...)" method is required to be split in order to become more readable.

**Review by illia-borysenko — APPROVED** [review]

**Comment by KVupp** [comment]
so the approach is
If invalid json is received - return 500 immediately
If unable to convert to xml - then skip the record?


is there a reason we are handling these differently?

**Comment by IlliaDaliek** [comment]
> so the approach is
> If invalid json is received - return 500 immediately
> If unable to convert to xml - then skip the record?
> 
> is there a reason we are handling these differently?
The reason is that if srs response is unparsable at all then we should not respond with a successful response. It means that data which oai-pmh is used is broken therefore we cannot process requests and we will face this in each  "List" request.

**Review by Igor-Gorchakov — APPROVED** [review]

File: src/main/java/org/folio/oaipmh/helpers/AbstractGetRecordsHelper.java [hunk 1]
```diff
@@ -144,58 +154,80 @@ private Response buildRecordsResponse(Context ctx, Request request,
...
           if (source != null && record.getHeader().getStatus() == null) {
             if (suppressedRecordsProcessingEnabled) {
               source = metadataManager.updateMetadataSourceWithDiscoverySuppressedData(source, instance);
             }
-            record.withMetadata(buildOaiMetadata(request, source));
+            try {
+              record.withMetadata(buildOaiMetadata(request, source));
+            } catch (Exception e) {
+              logger.error("Error occurred while converting record to xml representation.", e, e.getMessage());
+              logger.debug("Skipping problematic record due the conversion error. Source record id - " + recordId);
```
[the remark this answers is not in the document]
Comment (IlliaDaliek) [reply src/main/java/org/folio/oaipmh/helpers/AbstractGetRecordsHelper.java:179]: I don't think that this is needed. At such an error case the resumption token will not be useful to understand what happens. Just logging of record which caused the error is enough and brings all required information.

**Review by KVupp — COMMENTED** [review]
Just saw these. Just cross verify if these values are correct

File: .rancher-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+stages:
+  - name: Build
+    steps:
+      - runScriptConfig:
+          image: grizzlysoftware/maven-non-root:3.6.2-11.0.4-jdk-stretch
```
[the remark this answers is not in the document]
Comment (IlliaDaliek) [reply .rancher-pipeline.yml:5]: I took it from srs, we start the embedded Postgres during maven build and there were errors regarding permissions, so I just referred to srs as they use the same approach during the build step and this one setting helped here.

**Pull request merged**